src/app/nonprofit/nonprofit-page/donation-dialog.html

```diff
@@ -0,0 +1,124 @@
+                                <mat-label>Card Number</mat-label>
+                                <input formControlName="cardNumber" matInput type="tel" ccNumber>
+                                <mat-hint> Hint: Do NOT input real CC number</mat-hint>
+                            </mat-form-field>
+
+                            <!-- Card CVC-->
+                            <mat-form-field>
+                                <mat-label>CVC</mat-label>
+                                <input formControlName="cardCVC" matInput type="tel" ccCVC>
+                            </mat-form-field>
+                        </div>
+                    </div>
+                </div>
+                <div>
+                    <button mat-button matStepperPrevious>Back</button>
+                    <button mat-button matStepperNext>Go To Confirmation Page</button>
+                </div>
+                
+            </form>
+        </mat-step>
+
+        <!-- Confirmation Stepper -->
+        <mat-step>
+            <form [formGroup]="confirmationForm">
+                <ng-template matStepLabel>Confirm Your Donation</ng-template>
+
+                <div>
+                    <p>
+                        The amount you wish to donate and your payment information
+                        is displayed below. Please confirm your information and go back
+                        and make any changes should something be incorrect.
+                    </p>
+                    <div class="inputs-container">
+                        <div class="donation-input">
+                            <!-- Donation Amunt -->
+                            <p style="content: '$';"> 
+                                <b>Donation Amount: </b> ${{getDonationAmount()}} 
+                            </p>
+                            
+                            <!-- Card Holder Name -->
+                            <p> 
+                                <b>Card Holder Name:</b> {{getCardHolderName()}} 
+                            </p>
+
+                            <!-- Expiration Date-->
+                            <p> 
+                                <b>Expiration Date:</b> {{getCardExpDate()}}
+                            </p>
+                        </div>
+                    </div>
+                    <div class="inputs-container">
+                        <div class="donation-input">
+                            <!-- Card Number -->
+                            <p> 
+                                <b>Card Number:</b> {{getCardNumber()}} 
+                            </p>
+
+                            <!-- Card CVC-->
+                            <p>
+                                <b>Card CVC:</b> {{getCardCVC()}} 
+                            </p>
+                        </div>
+                    </div>
+                </div>
+
+                <button mat-button matStepperPrevious>Back</button>
+                <button mat-button >Submit Donation</button>
```

Comment:
[BUG] you mentioned this in the meeting already, but make sure the dialog closes and submits on submit and not on the x button close

Reply:
fixed